mflowgen/Tile_MemCore/constraints/constraints.tcl

```diff
@@ -36,11 +36,13 @@ set_driving_cell -no_design_rule \
 #
 # - make this non-zero to avoid hold buffers on input-registered designs
 
-set_input_delay -clock ${clock_name} [expr ${dc_clock_period}/2.0] [all_inputs]
+set_input_delay  -clock ${clock_name} [expr ${dc_clock_period}*0.8] [ all_inputs ]
+set_output_delay -clock ${clock_name} [expr ${dc_clock_period}*0.8] [ list read_config_data ]
+#set_input_delay  -clock ${clock_name} [expr ${dc_clock_period}*0.2] [ list chain_data_in chain_valid_in config_addr_in config_data_in config_en config_read config_write flush ren_in wen_in ]
+set_output_delay -clock ${clock_name} [expr ${dc_clock_period}*0.2] [ list hi lo config_out_write config_out_read config_out_config_data config_out_config_addr SB_T2_WEST_SB_OUT_B16 SB_T2_WEST_SB_OUT_B1 SB_T2_SOUTH_SB_OUT_B16 SB_T2_SOUTH_SB_OUT_B1 SB_T2_NORTH_SB_OUT_B16 SB_T2_NORTH_SB_OUT_B1 SB_T2_EAST_SB_OUT_B16 SB_T2_EAST_SB_OUT_B1 SB_T1_WEST_SB_OUT_B16 SB_T1_WEST_SB_OUT_B1 SB_T1_SOUTH_SB_OUT_B16 SB_T1_SOUTH_SB_OUT_B1 SB_T1_NORTH_SB_OUT_B16 SB_T1_NORTH_SB_OUT_B1 SB_T1_EAST_SB_OUT_B16 SB_T1_EAST_SB_OUT_B1 SB_T0_WEST_SB_OUT_B16 SB_T0_WEST_SB_OUT_B1 SB_T0_SOUTH_SB_OUT_B16 SB_T0_SOUTH_SB_OUT_B1 SB_T0_NORTH_SB_OUT_B16 SB_T0_NORTH_SB_OUT_B1 SB_T0_EAST_SB_OUT_B16 SB_T0_EAST_SB_OUT_B1]
```

Comment:
I know the timing constraints are still being worked out, but I'm confused as to how this could meet timing... We have 0.8 * clk_period input_delay for all inputs and 0.2 * clk_period output delay for many outputs (including all SB outputs). That's a whole clock period of external delay... does this meet timing? What's the rationale behind this?

Also it'd probably be good to use `get_ports` commands with wildcards instead of explicitly listing every port.

Reply:
These are changed